### PR TITLE
Fix IllegalArgumentException when ChestTerminal is requesting items

### LINKED ISSUE #3091: ChestTerminal causes IllegalArgumentException when there are empty slots in front of slots with requested items
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Due to the way ChestTerminal storage works a recent change to `ItemStackWrapper` uncovered a problem. Since the AccessTerminal looks for requested items in all slots of every inventory on the terminal channel, a newly added validation in the `ItemStackWrapper` constructor fails if an empty slot is found before the items requested.

### 📑 Reproduction Steps

1. Create a basic ChestTerminal setup - access terminal, cargo manager, and one chest with input and output nodes set to the terminal channel.
2. Insert a stack of any item into the chest's SECOND slot (keep the first slot empty)
3. Try to request the inserted item from the access terminal
4. IllegalArgumentException - see logs

### 💡 Expected Behavior

Items are delivered to the access terminal

### 📷 Screenshots / Videos

https://youtu.be/QrPyL12PTzk

### 📜 Server Log

https://pastebin.com/idkTbehi

### 📂 `/error-reports/` folder

Nothing, the exception is caught and logged properly

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.16.x

### ⭐ Slimefun version

DEV 927

### 🧭 Other plugins

ChestTerminal 43

### PR DESCRIPTION
**Pull request by martinbrom**
## Description

See attached issue

## Proposed changes

Check for null, we do not need to check empty slots.

## Related Issues (if applicable)

Resolves #3091

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by md5sha256** [comment]
Good catch! I missed that in my pr.

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoUtils.java [hunk 1]
```diff
@@ -185,8 +185,11 @@ static ItemStack withdrawFromVanillaInventory(AbstractItemNetwork network, Block
         for (int slot = minSlot; slot < maxSlot; slot++) {
             // Changes to these ItemStacks are synchronized with the Item in the Inventory
             ItemStack itemInSlot = contents[slot];
-            ItemStackWrapper wrapperInSlot = ItemStackWrapper.wrap(itemInSlot);
+            if (itemInSlot == null || itemInSlot.getType() == Material.AIR) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoUtils.java:188]:
```suggestion
            if (itemInSlot == null || itemInSlot.getType().isAir()) {
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
lgtm

**Review by WalshyDev — APPROVED** [review]
LGTM

**Pull request merged**